ros/src/waypoint_updater/waypoint_updater.py

```diff
@@ -26,26 +27,46 @@
 
 class WaypointUpdater(object):
     def __init__(self):
-        rospy.init_node('waypoint_updater')
-
-        rospy.Subscriber('/current_pose', PoseStamped, self.pose_cb)
-        rospy.Subscriber('/base_waypoints', Lane, self.waypoints_cb)
-
+	rospy.init_node('waypoint_updater')
+	self.ego_pos = 'None'
+	self.wps = 'None'
+	self.final_wps = 'None'
+	self.first_pass = True	
+        rospy.Subscriber('/current_pose', PoseStamped, self.pose_cb)	       
+	rospy.Subscriber('/base_waypoints', Lane, self.waypoints_cb)
         # TODO: Add a subscriber for /traffic_waypoint and /obstacle_waypoint below
-
-
         self.final_waypoints_pub = rospy.Publisher('final_waypoints', Lane, queue_size=1)
-
-        # TODO: Add other member variables you need below
-
         rospy.spin()
 
-    def pose_cb(self, msg):
-        # TODO: Implement
-        pass
+    def pose_cb(self, msg):        
+	self.ego_pos = msg.pose.position
+	if self.wps != 'None':	#Don't proceed until we have received waypoints
+		#return the index of the closest waypoint, given our current position (pose)
+		distances = []	
+		find_dist = lambda a, b: math.sqrt((a.x-b.x)**2 + (a.y-b.y)**2  + (a.z-b.z)**2)	
+		for i in range(len(self.wps.waypoints)):
+			#find the distance between each waypoint and the current position
+			distances.append(find_dist(self.wps.waypoints[i].pose.pose.position, self.ego_pos))
+	
+		#find index of waypoint closet to current position
+		closest_wp = distances.index(min(distances))
```

Comment:
I think we need to filter the waypoints to get only the ones in front of us. The first waypoint we return is the first one ahead of us according to the classroom.

Reply:
That's a good point, I think this will become more obvious once we try to get the car moving, which requires more than just the waypoint_updater.py to be completed.